aim/db/migration/alembic_migrations/versions/5e29ae4f45e6_host_domain_add_primary_key.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2017 Cisco Systems
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+"""Migrate HostDomainMapping to HostDomainMappingV2
+Revision ID: 5e29ae4f45e6
+Revises: 593d228d2fb4
+Create Date: 2017-10-25 16:39:33.713695
+"""
+
+# revision identifiers, used by Alembic.
+revision = '5e29ae4f45e6'
+down_revision = '593d228d2fb4'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.sql.expression import delete
+
+AIM_HOST_DOMAIN_MAPPING = 'aim_host_domain_mapping'
+AIM_HOST_DOMAIN_MAPPING_V2 = 'aim_host_domain_mapping_v2'
+
+# A model of the old domains table
+domains = sa.Table(AIM_HOST_DOMAIN_MAPPING, sa.MetaData(),
+                   sa.Column('host_name', sa.String(128), primary_key=True),
+                   sa.Column('vmm_domain_name', sa.String(64)),
+                   sa.Column('physical_domain_name', sa.String(64)))
+
+testing = True
+
+
+def upgrade():
+    session = sa.orm.Session(bind=op.get_bind())
+    if testing:
+        op.bulk_insert(domains, [{'host_name': 'host1',
+                                  'vmm_domain_name': 'foo',
+                                  'physical_domain_name': 'bar'}])
+    migrations = get_values(session, domains)
+
+    # A model of the new domains table
+    domainsv2 = op.create_table(
+        AIM_HOST_DOMAIN_MAPPING_V2,
+        sa.Column('host_name', sa.String(128)),
+        sa.Column('domain_name', sa.String(64)),
+        sa.Column('domain_type', sa.Enum('PhysDom',
+                                         'OpenStack',
+                                         'Kubernetes',
+                                         'VMware')),
+        sa.PrimaryKeyConstraint('host_name', 'domain_name', 'domain_type')
+    )
+
+    op.bulk_insert(domainsv2, migrations)
+    # we can get clear out the old table
+    delete_all = delete(domains)
+    session.execute(delete_all)
+
+
+def get_values(session, domains):
+    """Get pre-migration DB rows.
+
+    This gets the rows from the DB before the migration, using
```

Comment:
this comment is no longer true with this new v2 approach.

Reply:
You're correct. Will remove.